File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in s:
        if i == '(':
            stack.append(i)
        elif i == ')':
            if not stack:
                return False
            stack.pop()
            
        
    if len(stack) == 0:
        return True
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i=0; i<s.length(); i++) {
            char ch = s.charAt(i);
            if(stack.isEmpty()) {
                if(ch == '(') {
                    stack.push('(');
                } else {
                    return false;
                }
            } else {
                if(stack.peek() == '(') {
                    if(ch == ')') {
                        stack.pop();
                    } else {
                        stack.push('(');
                    }
                } else { // ')'
                    if(ch == ')') {
                        return false;
                    } else { // '('
                        stack.push('(');
                    }
                }
            }
        }
        if(stack.isEmpty()) {
            return true;
        } else {
            return false;
        }
    }
}